lib/bundler/diff.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Bundler
+  class Diff
+    def initialize(string1, string2)
+      @string1 = string1
+      @string2 = string2
+    end
+
+    def diff
+      @diff ||= begin
+          @paths = [tempfile(@string1), tempfile(@string2)]
+
+          cmd = format '"%s" %s %s', diff_bin, "-U10000", @paths.map {|s| %("#{s}") }.join(" ")
+          diff = `#{cmd}`
+          diff.force_encoding("ASCII-8BIT") if diff.respond_to?(:valid_encoding?) && !diff.valid_encoding?
+          diff
+        end
+    ensure
+      # unlink the tempfiles explicitly now that the diff is generated
+      if defined? @tempfiles # to avoid Ruby warnings about undefined ins var.
+        Array(@tempfiles).each do |t|
+          begin
+            # check that the path is not nil and file still exists.
+            # REE seems to be very agressive with when it magically removes
```

Comment:
REE is a very old implementation. I think we can just remove the file without any rescue or checks?

Reply:
Ruby Enterprise Edition was discontinued in 2012, which is 7 years ago now. I agree that we should not support REE in the next version of Bundler. 🙂